draft-ietf-rats-msg-wrap.md

```diff
@@ -234,6 +232,57 @@ A consumer can always distinguish tags that have been derived via
 tags that are not, and therefore apply the right decapsulation on
 receive.
 
+## CMW Collections {#cmw-coll}
+
+Layered Attesters and composite devices ({{Sections 3.2 and 3.3 of -rats-arch}}) generate Evidence that consists of multiple parts.
+
+For example, in data center servers, it is not uncommon for separate attesting environments (AE) to serve a subsection of the entire machine.
+One AE might measure and attest to what was booted on the main CPU, while another AE might measure and attest to what was booted on a SmartNIC plugged into a PCIe slot, and a third AE might measure and attest to what was booted on the machine's GPU.
+
+To address the composite Attester use case, this document defines a CMW "collection" as a container that holds several CMW items, each with a label that is unique within the scope of the collection.
+
+The CMW collection ({{fig-cddl-collection}}) is defined as a CBOR map or JSON object with CMW values, either native or "tunnelled" ({{cmw-tunnel}}).
+The position of a `cmw` entry in the `cmw-collection` is not significant.
+Instead, the labels identify a conceptual message that, in the case of a composite Attester, should typically correspond to a component of a system.
+Labels can be strings (or integers in the CBOR serialization) that serve as a mnemonic for different conceptual messages in the collection.
+Since the collection type is recursive, implementations may limit the allowed depth of nesting.
+
+~~~ cddl
+{::include cddl/cmw-collection.cddl}
+~~~
+{: #fig-cddl-collection artwork-align="left"
+   title="CDDL definition of the CMW collection format"}
+
+Although initially designed for the composite Attester use case, the CMW collection can be repurposed for other use cases requiring CMW aggregation.
+
+## CMW Tunnel {#cmw-tunnel}
+
+The CMW tunnel type ({{fig-cddl-tunnel}}) allows for moving a CMW in one serialization format, either JSON or CBOR, into a collection that uses the opposite serialization format.
+
+Both tunnel types are arrays with two elements.
+The first element, a fixed text string starting with a `#`, acts as a "magic" number.
+The `#`, which is not an acceptable start symbol for the `Content-Type` production ({{collected-cddl}}), makes it possible to disambiguate a CMW tunnel from a CMW array.
+
+~~~ cddl
+{::include cddl/cmw-tunnel.cddl}
+~~~
+{: #fig-cddl-tunnel artwork-align="left"
+   title="CDDL definition of the CMW tunnel format"}
+
+The conversion algorithms are described in the following sub sections.
```

Comment:
Inconsistent use of "subsection" and "sub section" within the same doc. I recommend the former.